plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergStatistics.java

```diff
@@ -0,0 +1,328 @@
+        private final Map<Integer, Long> columnSizes = new HashMap<>();
+        private long recordCount;
+        private long fileCount;
+        private long size;
+
+        public IcebergStatisticsBuilder(
+                Map<Integer, Type.PrimitiveType> idToTypeMapping,
+                List<Types.NestedField> columns,
+                TypeManager typeManager)
+        {
+            this.idToTypeMapping = requireNonNull(idToTypeMapping, "idToTypeMapping is null");
+            this.columns = requireNonNull(columns, "columns is null");
+            this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        }
+
+        public void registerDataFile(DataFile dataFile, PartitionSpec partitionSpec)
+        {
+            fileCount++;
+            recordCount += dataFile.recordCount();
+            size += dataFile.fileSizeInBytes();
+
+            Map<Integer, Long> newColumnSizes = dataFile.columnSizes();
+            if (newColumnSizes != null) {
+                for (Types.NestedField column : columns) {
+                    int id = column.fieldId();
+                    Long addedSize = newColumnSizes.get(id);
+                    if (addedSize != null) {
+                        columnSizes.merge(id, addedSize, Long::sum);
+                    }
+                }
+            }
+
+            Set<Integer> identityPartitionFieldIds = partitionSpec.fields().stream()
+                    .filter(field -> field.transform().isIdentity())
+                    .map(PartitionField::sourceId)
+                    .collect(toImmutableSet());
+            Map<Integer, Optional<String>> partitionValues = getPartitionKeys(dataFile.partition(), partitionSpec);
+            Map<Integer, Object> lowerBounds = convertBounds(idToTypeMapping, dataFile.lowerBounds());
+            Map<Integer, Object> upperBounds = convertBounds(idToTypeMapping, dataFile.upperBounds());
+            for (Types.NestedField column : columns) {
+                int id = column.fieldId();
+                io.trino.spi.type.Type trinoType = toTrinoType(column.type(), typeManager);
+                if (identityPartitionFieldIds.contains(id)) {
+                    Optional<String> partitionValue = partitionValues.get(id);
+                    if (partitionValue.isPresent()) {
+                        Type type = idToTypeMapping.get(id);
+                        Object trinoValue = deserializePartitionValue(toTrinoType(type, typeManager), partitionValue.get(), column.name());
+                        // Update min/max stats but there are no null values to count
+                        updateMinMaxStats(
+                                id,
+                                trinoType,
+                                trinoValue,
+                                trinoValue,
+                                Optional.of(0L),
+                                dataFile.recordCount());
+                        updateNullCountStats(id, Optional.of(0L));
```

Comment:
This shouldn't change state, should it? You can skip this line?

Reply:
It does. Knowing that a row has no null values is represented as Optional.of(0L), not knowing anything about the null count is Optional.empty(). I can add some javadoc around that